Carbon.Redis/Abstractions/ICarbonCacheExtensions.cs

```diff
@@ -83,39 +127,164 @@ public static async Task HashSetAsObject<T>(this ICarbonCache instance, string k
             return JsonSerializer.Deserialize<T>(JsonSerializer.Serialize(keyValuePairs));
         }
 
+        /// <summary>
+        /// Gets your entire object with the given fields that is converted into hash key-value pairs within a single redis hash key, if key does not exist, it tries to fill with the given function
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="instance"></param>
+        /// <param name="key">Redis Key</param>
+        /// <param name="setMethodIfNotExists">Execute if this key does not exist</param>
+        /// <returns>Your entire object</returns>
+        public static async Task<T> HashGetAsObject<T>(this ICarbonCache instance, string key, Func<Task<T>> setMethodIfNotExists) where T : class
+        {
+            var value = await instance.HashGetAsObject<T>(key);
+
+            if(value == default)
+            {
+                T result = await setMethodIfNotExists();
```

Comment:
Just a suggestion, but maybe we should call the user-supplied method in a try-catch block. This way user error won't interfere with our inner workings.

Reply:
Maybe we can have a special exception type for erroneous functions given by the users

Reply:
Users are fully responsible of the error handling, if they do not obey, they will accept the consequences :)